cmd/query/app/builder/storage.go

```diff
@@ -43,22 +42,21 @@ var (
 )
 
 // NewStorageBuilder creates a StorageBuilder based off the flags that have been set
-func NewStorageBuilder(opts ...basicB.Option) (StorageBuilder, error) {
-	flag.Parse()
+func NewStorageBuilder(parsedFlags *flags.Flags, opts ...basicB.Option) (StorageBuilder, error) {
```

Comment:
This PR is getting to the mess I will do it in the follow up PR here is the issue https://github.com/uber/jaeger/issues/296